atoma-proxy/src/server/streamer.rs

```diff
@@ -618,9 +615,10 @@ impl Drop for Streamer {
                 if let Err(e) = update_state_manager_fiat(
                     &self.state_manager_sender,
                     self.user_id,
-                    self.fiat_estimated_amount.unwrap_or_default(),
-                    self.num_generated_tokens * self.price_per_million.unwrap_or_default()
-                        / ONE_MILLION as i64,
+                    self.estimated_total_tokens,
+                    self.num_generated_tokens,
+                    self.price_per_million.unwrap_or_default(),
```

Comment:
unwrap or default ? How come ? Default will be 0..

Reply:
ok, now the price per million always filled